src/client/AM.PA.MacOs/PersonalAnalytics/App/AppDelegate.swift

```diff
@@ -280,10 +301,47 @@ class AppDelegate: NSObject, NSApplicationDelegate, NSUserNotificationCenterDele
 
     
     func userNotificationCenter(_ center: NSUserNotificationCenter, didActivate notification: NSUserNotification) {
-        //print("Using delelgate for NSUsernotification")
-        self.toggleSummary(notification.self)
+
```

Comment:
I tried to move as much code as I could to my tracker's files. 
I need a way however to pass each activated notification to the `EmotionTracker` class. Its `manageNotification` method verifies the ownership of the notification and manages its actions when appropriate. I couldn't find a way to do this without touching the AppDelegate.

In addition, the `SummaryView` of the `TaskProductivityTracker` always shows up when a notification is clicked (even when it was scheduled by `EmotionTracker`). I had to comment out line 284:

> `self.toggleSummary(notification.self)`

to deal with this. 
A solution should be found here in order to know which tracker scheduled the notification and delegate it to respond. I ask your advice about this problem, since I can't find a way to solve it without making edits to your code.

Reply:
I feel like there must be a better solution than this, but I dont know what it is either 🙂. For the time being, I think if you just add a switch statement on the notification identifier it would be ok i.e.
```suggestion
userNotificationCenter(_ center: NSUserNotificationCenter, didActivate notification: NSUserNotification){
  switch(notification.identifier){
    case <your identifier>:
      ...
    ...
  }
}
```

Just add the identifier as a constant in the `AppConstants.swift` file.

It's fine if you leave that line commented out for now as that tracker is a work in progress anyways

Reply:
Right, thanks for the tip 🙂